src/client/cli/cmd/launch.cpp

```diff
@@ -471,3 +443,83 @@ mp::ReturnCode cmd::Launch::request_launch(const ArgParser* parser)
 
     return dispatch(&RpcMethod::launch, request, on_success, on_failure, streaming_callback);
 }
+
+auto cmd::Launch::ask_metrics_permission(const mp::LaunchReply& reply) -> OptInStatus::Status
+{
+    if (term->is_live())
+    {
+        cout << "One quick question before we launch … Would you like to help\n"
+             << "the Multipass developers, by sending anonymous usage data?\n"
+             << "This includes your operating system, which images you use,\n"
+             << "the number of instances, their properties and how long you use them.\n"
+             << "We’d also like to measure Multipass’s speed.\n\n"
+             << (reply.metrics_show_info().has_host_info() ? "Choose “show” to see an example usage report.\n\n"
+                                                             "Send usage data (yes/no/Later/show)? "
+                                                           : "Send usage data (yes/no/Later)? ");
+
+        while (true)
+        {
+            std::string answer;
+            std::getline(term->cin(), answer);
+            if (std::regex_match(answer, yes))
+            {
+                cout << "Thank you!\n";
+                return OptInStatus::ACCEPTED;
+            }
+            else if (std::regex_match(answer, no))
+            {
+                return OptInStatus::DENIED;
+            }
+            else if (answer.empty() || std::regex_match(answer, later))
+            {
+                return OptInStatus::LATER;
+            }
+            else if (reply.metrics_show_info().has_host_info() && std::regex_match(answer, show))
+            {
+                // TODO: Display actual metrics data here provided by daemon
+                cout << "Show metrics example here\n\n"
+                     << "Send usage data (yes/no/Later/show)? ";
+            }
+            else
+            {
+                cout << (reply.metrics_show_info().has_host_info() ? "Please answer yes/no/Later/show: "
+                                                                   : "Please answer yes/no/Later: ");
+            }
+        }
+    }
+
+    return OptInStatus::UNKNOWN;
+}
+
+bool cmd::Launch::ask_bridge_permission(multipass::LaunchReply& reply)
+{
+    static constexpr auto plural = "Multipass needs to create {} to connect to {}.\nThis will temporarily disrupt "
+                                   "connectivity on those interfaces.\n\nDo you want to continue (yes/no)?";
+    static constexpr auto singular = "Multipass needs to create a {} to connect to {}.\nThis will temporarily disrupt "
+                                     "connectivity on that interface.\n\nDo you want to continue (yes/no)?";
```

Comment:
```suggestion
    static constexpr auto plural = "Multipass needs to create {} to connect to {}.\nThis will temporarily disrupt "
                                   "connectivity on those interfaces.\n\nDo you want to continue (yes/no)? ";
    static constexpr auto singular = "Multipass needs to create a {} to connect to {}.\nThis will temporarily disrupt "
                                     "connectivity on that interface.\n\nDo you want to continue (yes/no)? ";
```

Don't want to make `Yes` the default?

Reply:
We could. Personally I thought the inconvenience was worth it here, to impress the (potential) weight of the decision on the user (who knows whatever is going on that depends on networking). But it's not a strong preference. 

Let me know if you think the default would be best. We can also ask others what they think.

Reply:
Note that the other question doesn't have a default either. There I guess we could safely set it to "Later".

Reply:
Right, "Later" is not really a default, is it ;)

Reply:
I am not sure I get what you mean. Syntactically, it is one of the possible replies. Semantically, "no decision yet"  is arguably the best default decision :)

Reply:
> arguably the best default decision

I'm thinking: the one that assumes the least, so has the least compromise.

Reply:
Yeah, good as is.